bot/code_review_bot/report/phabricator.py

```diff
@@ -79,28 +63,37 @@ def publish(self, issues, revision, task_failures):
             for patch in revision.improvement_patches
             if patch.analyzer not in self.analyzers_skipped
         ]
-        # List of issues without possible build errors
-        issues_only = [issue for issue in issues if not issue.is_build_error()]
-        build_errors = [issue for issue in issues if issue.is_build_error()]
-
-        if issues_only or build_errors or task_failures:
 
-            # Always publish comment summarizing issues
-            self.publish_comment(revision, issues_only, patches, task_failures)
-
-            # Publish all errors as lint issues
-            if self.publish_errors:
-                lint_issues = [
-                    issue for issue in issues_only if issue.level == Level.Error
-                ]
-            else:
-                lint_issues = []
+        if issues or task_failures:
+            # Split issues and build errors
+            issues_only = [issue for issue in issues if not issue.is_build_error()]
+            build_errors = [issue for issue in issues if issue.is_build_error()]
+            non_coverage_issues = [
+                issue for issue in issues_only if not isinstance(issue, CoverageIssue)
+            ]
+            coverage_issues = [
+                issue for issue in issues_only if isinstance(issue, CoverageIssue)
+            ]
+
+            if non_coverage_issues or build_errors:
+                # Publish on Harbormaster all at once
+                # * All non coverage publishable issues as lint issues
+                # * All build errors as unit test results
+                self.publish_harbormaster(revision, non_coverage_issues, build_errors)
+
+            if non_coverage_issues or patches or task_failures:
+                # Publish comment summarizing issues
```

Comment:
Do we still want the summary comment now that we have all issues as lint issues?

Reply:
Decided on Matrix: yes, because we want the email notification associated with the comment. We can remove it at some point, filed #551 as a follow-up.